cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -465,5 +465,32 @@ type ComponentConfig interface {
 	Get() interface{}
 }
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ResetConfiguration contains a list of elements that is specific "kubeadm reset"-only runtime information.
+type ResetConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// SkipPhases is a list of phases to skip during command execution.
+	// The list of phases can be obtained with the "kubeadm reset phase --help" command.
+	SkipPhases []string
+
+	// Force flag instructs kubeadm to reset the node without prompting for confirmation
+	Force bool
+
+	// CertificatesDir specifies where to store or look for all required certificates.
+	// Cannot mix the config for both the resetConfig and flag to make it consistent with kubeadm init.
+	CertificatesDir string
+
+	// CRISocket is used to retrieve container runtime info. This information will be annotated to the Node API object, for later re-use.
```

Comment:
> This information will be annotated to the Node API object, for later re-use.

This description is fine for `Node Registration`, but not suitable for `Node Reset`.

Reply:
Done